### PR TITLE
Add woodpecker support

### PR DESCRIPTION
**Pull request by 6543**
https://woodpecker-ci.org

and also refactor detection to be more flexible and make code more readable

### CODE REVIEW

**Comment by 6543** [comment]
-> https://github.com/watson/ci-info/pull/90

**Comment by 6543** [comment]
@adamdecaf tell me if i should split this into different pulls

**Comment by 6543** [comment]
well you dont have to wait for hacktoberfest ... I'll have at least 10 pulls there too :D
normal tendency is from 20-30pulls

File: .woodpecker.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+pipeline:
+  test:
+    image: golang:1.12
+    commands:
+      - go vet ./...
+      - go test ./...
+    environment:
+      - CURRENT_CI_PROVIDER=Woodpecker CI
```
Comment (6543) [inline .woodpecker.yml:8]: newline prefered?

Comment (adamdecaf) [reply .woodpecker.yml:8]: Nah, that's fine.

**Comment by adamdecaf** [comment]
Thanks for opening this! I've got to refresh on this project a bit.

**Comment by 6543** [comment]
thanks :tada:

**Pull request merged**